feat: enable colored caption bar on Windows

mao-yining opened the pull request:
Problem: gVim on Windows' caption bar is always white.

Solution: add gui_mch_set_caption() to cope with that.

The implement of 'guidarkmode' is a much longer task, so I would like to complete this most needed feature.

This commit seen the caption bar as the extension of the background so it is follow the option 'background' before Windows 11 and follow the background color after Windows 11.

Closes: #3922, #18028

mao-yining committed:
feat: enable colored caption bar on Windows

Problem: gVim on Windows' caption bar is always white.

Solution: add gui_mch_set_caption() to cope with that.

The implement of 'guidarkmode' is a much longer task, so I would like to
complete this most needed feature.

This commit seen the caption bar as the extension of the background so
it is follow the option 'background' before Windows 11 and follow the
background color after Windows 11.

Closes: #3922, #18028

Signed-off-by: Mao-Yining <mao.yining@outlook.com>

chrisbra commented on runtime/doc/gui_w32.txt:
```diff
@@ -488,4 +488,12 @@ VIM_KEYCODE_TRANS_STRATEGY can be set to the desired value ("experimental" or
         gvim.exe
 <
+
+Keycode translation strategy			*w32-caption-bar-setting*
```
That title seems to be misleading

chrisbra commented on runtime/doc/gui_w32.txt:
```diff
@@ -488,4 +488,12 @@ VIM_KEYCODE_TRANS_STRATEGY can be set to the desired value ("experimental" or
+
+The color of caption bar is seem as an extension of the background.
+So before Windows 11 the color will obey the value of 'background',
+and after Windows 11 the color will obey the value of |:hi-normal|,
```
I suppose you mean:
```suggestion
and starting with Windows 11 the color will obey the value of |:hi-normal|,
```

mao-yining committed:
Docs: Modified gui_w32.txt

Signed-off-by: Mao-Yining <mao.yining@outlook.com>

mao-yining committed:
Fix: some tested problem

Signed-off-by: Mao-Yining <mao.yining@outlook.com>

mao-yining committed:
Regenerate Helptags

Signed-off-by: Mao-Yining <mao.yining@outlook.com>

mao-yining committed:
Fix: tested problem about tags

It seems :helptags has different behaviour with test's setting

Signed-off-by: Mao-Yining <mao.yining@outlook.com>

mao-yining committed:
Revert "Fix: tested problem about tags"

This reverts commit 35ad6685d8ad50e01fd98b099f27cad01d65dbd0.

mao-yining committed:
Use doctags.vim build tags

Signed-off-by: Mao-Yining <mao.yining@outlook.com>

mao-yining committed:
Some Style Change

Signed-off-by: Mao-Yining <mao.yining@outlook.com>

chrisbra commented on runtime/doc/gui_w32.txt:
```diff
@@ -488,4 +488,13 @@ VIM_KEYCODE_TRANS_STRATEGY can be set to the desired value ("experimental" or
         gvim.exe
 <
+
+Caption Bar's Behaviour			    *gui-w32-caption-bar-behaviour*
```
I had to look up the term caption bar. I think title bar is more common, so I'd prefer to use that one.

mao-yining replied:
You can decide it when merging, I choose this because Microsoft use "caption" in its documents.

mao-yining committed:
Remove Support for Windows 10

Due to its the 'background' is default dark on Windows, which is
disharmony with current default colorscheme.

Signed-off-by: Mao-Yining <mao.yining@outlook.com>

mao-yining committed:
Update the document

Signed-off-by: Mao-Yining <mao.yining@outlook.com>
